hazelcast/src/main/java/com/hazelcast/query/impl/InternalIndex.java

```diff
@@ -118,10 +121,19 @@ public interface InternalIndex extends Index {
 
     /**
      * @return Step-aware storage that backs the Index.
-     * By default returns {@code null} that indicates there is no
-     * Step-aware backed storage.
+     * By default, returns {@code null} that indicates
+     * there is no step-aware backed storage.
      */
     default StepAwareStorage getStepAwareStorage() {
         return null;
     }
+
+    /**
+     * @param isDescending whether the index is used in descending order.
+     * @return             comparator ordering IMap keys stored for given index key
+     */
+    @Nullable
+    default Comparator getComparator(boolean isDescending) {
+        return null;
```

Comment:
I don't know what is `FirstComponentDecorator` and `QueryTrackingIndex` but invocation there should probably be delegated. Remove default implementation from `Index` to avoid such mistakes.

Reply:
https://github.com/hazelcast/hazelcast-mono/pull/52/commits/c2b71cd2bc5489bd928ba6224fd896132797f66b

Reply:
✔️